tools/link.py

```diff
@@ -2128,6 +2128,18 @@ def create_esm_wrapper(wrapper_file, support_target, wasm_target):
     wrapper.append(f"export {{ default, {js_exports} }} from '{support_url}';")
   else:
     wrapper.append(f"export {{ default }} from '{support_url}';")
+
+  if settings.ENVIRONMENT_MAY_BE_NODE and settings.EXPECT_MAIN:
```

Comment:
Doesn't need to be done in this PR, but maybe gate this on INVOKE_RUN in the future?

Reply:
Done